APIs/SinkMetadataProcessingAPI.raml

```diff
@@ -208,21 +208,96 @@ mediaType: application/json
               type: !include schemas/error.json
     /edid:
       get:
-        description: >
-          Redirects to the location of the effective EDID file or returns the effective EDID file directly.
-          Check HTTP response code to determine which mode of operation is in use. When returning the effective EDID file
-          directly ensure that the `Content-Type` entity header is correctly set.
+        description: List all the API endpoints for EDIDs associated with the Input.
         responses:
           200:
-            description: The effective EDID file itself will be served
+            description: Returned when success.
             body:
-              application/octet-stream:
-                type: file
-          307:
-            description: A temporary redirect to the effective EDID file
+              example: !include ../examples/input-edid-get-200.json
+              type: !include schemas/input-edid-base.json
+          204:
+            description: The upstream counterpart of this Input does not support EDID.
           404:
-            description: >
-              Returned when the effective EDID file or the requested Input ID doesn't exist.
+            description: Returned when the requested Input ID does not exist.
+            body:
+              type: !include schemas/error.json
+      /base:
+        get:
+          description: >
+            Redirects to the location of the Base EDID file or returns the Base EDID file directly.
+            Check HTTP response code to determine which mode of operation is in use. When returning the Base EDID file
+            directly ensure that the `Content-Type` entity header is correctly set.
+          responses:
+            200:
+              description: The Base EDID file itself.
+              body:
+                application/octet-stream:
+                  type: file
+            204:
+              description: The upstream counterpart of this Input does not support EDID.
+              body:
+                type: !include schemas/error.json
+            307:
+              description: A temporary redirect to the Base EDID file.
+            404:
+              description: >
+                Returned when the requested Input ID doesn't exist.
+              body:
+                type: !include schemas/error.json
+        put:
+          description: Replace Base EDID with new one
+          responses:
+            202:
```

Comment:
I think that 200 is more appropriate. The base EDID has been accepted by the device and becomes the current EDID. Are you returning 202 because the new base EDID will not be used until new constraints are PUT?

Reply:
It makes sense to me to use Base EDID right after accepting it.

https://datatracker.ietf.org/doc/html/rfc7231#section-6.3.1says "Aside from responses to CONNECT, a 200 response always has a payload, though an origin server MAY generate a payload body of zero length". I don't know if it makes sense to put the accepted Base EDID binary to the payload so I chose 202 Accepted with no body.

Reply:
Why not 204 then?

Reply:
I replaced 202 with 204 so that if an Input/Output doesn't support/provide EDID, then all GETs under `/edid` return `204 No Content`, `PUT /edid/base` returns `405 Method Not Allowed` and `edid_support` is `false` in `/properties`. If an Input supports EDID but doesn't support setting arbitrary Base EDID, then `PUT /edid/base` returns `405 Method Not Allowed`. I assume this case is rare enough to signalize it with an additional boolean.

Reply:
2021/11/22 activity group call: marked as resolved.